packages/eslint-plugin/typings/eslint-rules.d.ts

```diff
@@ -817,17 +817,60 @@ declare module 'eslint/lib/rules/no-invalid-this' {
     ],
     {
       // for ESLint < v8.7.0
+
       Program?: (node: TSESTree.Program) => void;
       'Program:exit'?: (node: TSESTree.Program) => void;
+
       FunctionDeclaration?: (node: TSESTree.FunctionDeclaration) => void;
       'FunctionDeclaration:exit'?: (node: TSESTree.FunctionDeclaration) => void;
+
       FunctionExpression?: (node: TSESTree.FunctionExpression) => void;
       'FunctionExpression:exit'?: (node: TSESTree.FunctionExpression) => void;
 
       // for ESLint >= v8.7.0
-      // We don't use it and we don't have the CodePath types, so comment out it.
-      // onCodePathStart?: (codePath: unknown, node: TSESTree.Node) => void
-      // onCodePathEnd?: (codePath: unknown, node: TSESTree.Node) => void
+
+      // The code path functions below are intentionally commented out because
+      // they cause unresolvable compiler errors:
+      // https://github.com/typescript-eslint/typescript-eslint/issues/6993
+      // Note that plugin authors can copy-paste these functions into their own
+      // code as selectors and they will still work as long as the second argument
+      // is omitted.
+
+      /*
+      onCodePathStart?: (
+        codePath: TSESLint.CodePath,
+        node: TSESTree.Node,
+      ) => void;
+      */
+
+      /*
+      onCodePathEnd?: (
+        codePath: TSESLint.CodePath,
+        node: TSESTree.Node,
+      ) => void;
+      */
+
+      /*
+      onCodePathSegmentStart?: (
+        segment: TSESLint.CodePathSegment,
+        node: TSESTree.Node,
+      ) => void;
+      */
+
+      /*
+      onCodePathSegmentEnd?: (
+        segment: TSESLint.CodePathSegment,
+        node: TSESTree.Node,
+      ) => void;
+      */
+
+      /*
+      onCodePathSegmentLoop?: (
+        fromSegment: TSESLint.CodePathSegment,
+        toSegment: TSESLint.CodePathSegment,
+        node: TSESTree.Node,
+      ) => void;
+      */
```

Comment:
This particular file (`packages/eslint-plugin/typings/eslint-rules.d.ts`) is private and internal to our repo - it isn't included in any releases or even really intended for viewing by anyone.

I'm more than happy for these comments to exist! But they should be put in a place that consumers can actually see them. So probably within [this currently empty interface?](https://github.com/IsaacScript/typescript-eslint/blob/interfaces/packages/utils/src/ts-eslint/Rule.ts#L553)

Could you please revert changes to this file and move the comments into the other file?

once that's done we can land this!